bin/markdownlint-all

```diff
@@ -1,6 +1,6 @@
-#!/bin/bash
+#!/bin/sh -eu
 
-bindir=$( cd "${BASH_SOURCE[0]%/*}" && pwd )
+bindir=$( cd "${0%/*}" && pwd )
```

Comment:
Could this cause problems if someone adds `bin` to their PATH?

E.g.

```sh
export PATH="$PWD/bin:$PATH"
markdownlint-all
```

Reply:
> Could this cause problems if someone adds `bin` to their PATH?

I did try that, in both bash and dash, and it behaves exactly the same if I use relative or absolute path, or calling `markdownlint-all` with the `linkerd2/bin` directory in `PATH`. Also regardless of which directory I am currently in when calling the command.

This behavior is also the same as what I get with the bash-specific scripts currently on master.

Reply:
Thanks!